flow/flow_ebos_foam.cpp

```diff
@@ -43,19 +43,19 @@ struct EnableFoam<TypeTag, TTag::EclFlowFoamProblem> {
 }}
 
 namespace Opm {
-void flowEbosFoamSetDeck(double setupTime, std::unique_ptr<Deck> deck,
-                         std::unique_ptr<EclipseState> eclState,
-                         std::unique_ptr<Schedule> schedule,
-                         std::unique_ptr<SummaryConfig> summaryConfig)
+void flowEbosFoamSetDeck(double setupTime, std::shared_ptr<Deck> deck,
+                         std::shared_ptr<EclipseState> eclState,
+                         std::shared_ptr<Schedule> schedule,
+                         std::shared_ptr<SummaryConfig> summaryConfig)
 {
     using TypeTag = Properties::TTag::EclFlowFoamProblem;
     using Vanguard = GetPropType<TypeTag, Properties::Vanguard>;
 
     Vanguard::setExternalSetupTime(setupTime);
-    Vanguard::setExternalDeck(std::move(deck));
-    Vanguard::setExternalEclState(std::move(eclState));
-    Vanguard::setExternalSchedule(std::move(schedule));
-    Vanguard::setExternalSummaryConfig(std::move(summaryConfig));
+    Vanguard::setExternalDeck(deck);
+    Vanguard::setExternalEclState(eclState);
+    Vanguard::setExternalSchedule(schedule);
+    Vanguard::setExternalSummaryConfig(summaryConfig);
```

Comment:
Technically, these could still be `std::move()`s.  The by-value parameter to `flowEbosFoatSetDeck()` will increase the `shared_ptr<>`'s use-count by one and that's all we really need here.  Passing the pointers on by value will then increase the use-count by one more&mdash;meaning the use-count is temporarily increased by a total of two&mdash;until `flowEbosSetDeck()` returns.